projects/plugins/jetpack/_inc/lib/icalendar-reader.php

```diff
@@ -458,44 +553,54 @@ protected function filter_past_and_recurring_events( $events ) {
 	/**
 	 * Parse events from an iCalendar feed
 	 *
-	 * @param string $url (default: '')
+	 * @param string $url (default: '').
 	 * @return array | false on failure
 	 */
 	public function parse( $url = '' ) {
-		$cache_group = 'icalendar_reader_parse';
+		$cache_group     = 'icalendar_reader_parse';
 		$disable_get_key = 'disable:' . md5( $url );
 
-		// Check to see if previous attempts have failed
-		if ( false !== wp_cache_get( $disable_get_key, $cache_group ) )
+		// Check to see if previous attempts have failed.
+		if ( false !== wp_cache_get( $disable_get_key, $cache_group ) ) {
 			return false;
+		}
 
-		// rewrite webcal: URI schem to HTTP
-		$url = preg_replace('/^webcal/', 'http', $url );
-		// try to fetch
-		$r = wp_remote_get( $url, array( 'timeout' => 3, 'sslverify' => false ) );
+		// rewrite webcal: URI schem to HTTP.
+		$url = preg_replace( '/^webcal/', 'http', $url );
+		// try to fetch.
+		$r = wp_remote_get(
+			$url,
+			array(
+				'timeout'   => 3,
+				'sslverify' => false,
+			)
+		);
 		if ( 200 !== wp_remote_retrieve_response_code( $r ) ) {
-			// We were unable to fetch any content, so don't try again for another 60 seconds
+			// We were unable to fetch any content, so don't try again for another 60 seconds.
 			wp_cache_set( $disable_get_key, 1, $cache_group, 60 );
 			return false;
 		}
 
 		$body = wp_remote_retrieve_body( $r );
-		if ( empty( $body ) )
+		if ( empty( $body ) ) {
 			return false;
+		}
 
-		$body = str_replace( "\r\n", "\n", $body );
+		$body  = str_replace( "\r\n", "\n", $body );
 		$lines = preg_split( "/\n(?=[A-Z])/", $body );
 
-		if ( empty( $lines ) )
+		if ( empty( $lines ) ) {
 			return false;
+		}
 
-		if ( false === stristr( $lines[0], 'BEGIN:VCALENDAR' ) )
+		if ( false === stristr( $lines[0], 'BEGIN:VCALENDAR' ) ) {
 			return false;
+		}
 
 		foreach ( $lines as $line ) {
-			$add  = $this->key_value_from_string( $line );
+			$add = $this->key_value_from_string( $line );
 			if ( ! $add ) {
-				$this->add_component( $type, false, $line );
+				$this->add_component( '', false, $line );
```

Comment:
This changes behavior, and probably in a bad way. Before this would append lines in between "BEGIN:VEVENT" and "END:VEVENT" to a "VEVENT" component. Now it'll leave the VEVENT component empty and shove everything into an empty-string component instead.

To keep the current behavior, you'd want to set `$type = ''` before line 600 and leave this alone.

Reply:
👍 Done in 142e0159190d67bc3bb8d771ab2eb773e88bcdf9